File: backend/src/main/java/com/example/demo/models/Project.java
package com.example.demo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "projects")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    @Column(columnDefinition = "TEXT")
    private String description;
    private String githubUrl;
    private String liveUrl;
    private String imageUrl; // cloudinary link
    private String role;
    private String stackSummary;
    private String projectType;
    private String status;
    private LocalDate startedAt;
    private LocalDate finishedAt;
    private boolean isPinned = false;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;
}

File: frontend/src/types/project.ts
export interface ProjectApi {
  id: number;
  title: string;
  description: string;
  githubUrl?: string | null;
  liveUrl?: string | null;
  imageUrl?: string | null;
  role?: string | null;
  stackSummary?: string | null;
  projectType?: string | null;
  status?: string | null;
  startedAt?: string | null;
  finishedAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  pinned?: boolean;
  isPinned?: boolean;
}

export interface Project {
  id: number;
  title: string;
  description: string;
  githubUrl?: string | null;
  liveUrl?: string | null;
  imageUrl?: string | null;
  role?: string | null;
  stackSummary?: string | null;
  projectType?: string | null;
  status?: string | null;
  startedAt?: string | null;
  finishedAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  isPinned: boolean;
}

export interface ProjectPayload {
  title: string;
  description: string;
  githubUrl?: string;
  liveUrl?: string;
  imageUrl?: string;
  role?: string;
  stackSummary?: string;
  projectType?: string;
  status?: string;
  startedAt?: string;
  finishedAt?: string;
  pinned: boolean;
}

export interface GithubRepoInfo {
  fullName: string;
  name: string;
  description: string;
  htmlUrl: string;
  stars: number;
  forks: number;
  language: string | null;
  ownerAvatarUrl: string | null;
}
